example/conf.py

```diff
@@ -74,6 +74,8 @@
     # Controls the existence of Previous / Next buttons at the bottom of pages
     # Valid options: none, prev, next, both
     "sequential_nav": "none",
+    # Controls whether to display the contributors for each file
+    "display_contributors": False,
```

Comment:
Actually I think a better fix would be to leave the default setting (to show default contributors), add the project to a fake repository with a fake author and then check that the generated html has the "Thanks for Jane Doe for the contribution" (well, the actual wording). That way we know that the extension is correctly enabled and we protect ourselves a bit from future regressions

Reply:
I was considering that - but (apart from the fact that this might challenge my coding skills :wink: ) is this extension really the right place to test functionality from other Sphinx extensions?

I've just opened a PR to add testing to the starter pack: https://github.com/canonical/sphinx-docs-starter-pack/pull/261
I think this might be the better place for such integration tests.

Reply:
In a "purist" sense I agree with you, but in a "pragmatic" sense I'm thinking about the maintenance story here. 

This extension is basically "glue" for a lot of other extensions, some of which we maintain and some of which we don't, and if at some point an user opens a bug saying "contributor listing is not showing up" it's very helpful to have a test to play around with and ensure that the extension is working as expected in the base case.

It's also helpful to detect regressions as those other extensions release updates, so we can do informed restrictions on version ranges and such.

Finally, the fact that we enable all of these extensions at the same time means that we're the best ones to check that they work as expected *together*, as opposed to the isolated tests that the individual extensions have/should have.

I can take a crack at the coding side of it if you want; I might do it anyway (add the test to a blank git repo at test-time) just because it's more representative of a typical project.

Reply:
Yeah, I see your point, it sounds like a good idea.

> I can take a crack at the coding side of it if you want; I might do it anyway (add the test to a blank git repo at test-time) just because it's more representative of a typical project.

That would be great, thanks - I feel I would be wasting a lot of time just setting everything up on my end and figuring out where to start.

Reply:
then I'll do it in a follow up and we can merge this in the meantime